packages/flame/lib/src/components/input/advanced_button_component.dart

```diff
@@ -0,0 +1,270 @@
+import 'package:flame/components.dart';
+import 'package:flame/events.dart';
+import 'package:flutter/foundation.dart';
+
+class AdvancedButtonComponent extends PositionComponent
+    with HoverCallbacks, TapCallbacks {
+  AdvancedButtonComponent({
+    PositionComponent? defaultSkin,
+    PositionComponent? downSkin,
+    PositionComponent? hoverSkin,
+    this.onPressed,
+    bool isSelectable = false,
```

Comment:
These two should either move up or down, so that all the skins are together

Reply:
<img width="427" alt="image" src="https://github.com/flame-engine/flame/assets/18004353/d8c15be1-e725-4d7b-826d-6669a620a965">
Better this way